lib/solr_wrapper/checksum_validator.rb

```diff
@@ -27,13 +27,19 @@ def validate!(file)
     private
 
       def checksumurl(suffix)
-        if config.default_download_url == config.static_config.archive_download_url
-          "#{config.default_download_url}.#{suffix}"
+        if remote_checksum?(suffix)
+          config.checksum
+        elsif config.mirror_url != config.archive_download_url
```

Comment:
I don't think we ever want to trust the checksum from a mirror if the goal is to make sure the mirror hasn't maliciously tampered with the files.

Reply:
Ok. My use case was "apache.org is down, so switch to a mirror", which without that would require me to additionally set an explicit checksum URL, or ignore_checksum.